code/datums/gamemodes/game_mode.dm

```diff
@@ -860,3 +868,86 @@ Sensors indicate [numXenosShip ? "[numXenosShip]" : "no"] unknown lifeform signa
 		return FALSE
 
 	return new_xeno
+
+/datum/game_mode/proc/set_valid_job_types()
+	if(!SSjob?.initialized)
+		to_chat(world, "<span class='boldnotice'>Error setting up valid jobs, no job subsystem found initialized.</span>")
+		CRASH("Error setting up valid jobs, no job subsystem found initialized.")
+	if(!length(valid_job_types))
+		SSjob.active_occupations = SSjob.occupations
+		return TRUE
+	SSjob.active_occupations.Cut()
+	for(var/j in SSjob.occupations)
+		var/datum/job/job = j
+		if(!valid_job_types[job.type])
+			job.total_positions = 0
+			continue
+		job.total_positions = valid_job_types[job.type]
+		SSjob.active_occupations += job
+	if(!length(SSjob.active_occupations))
+		to_chat(world, "<span class='boldnotice'>Error, game mode has only invalid jobs assigned.</span>")
+		return FALSE
+	return TRUE
+
+/datum/game_mode/proc/set_valid_squads()
+	var/max_squad_num = min(squads_max_number, SSmapping.configs[SHIP_MAP].squads_max_num)
+	if(max_squad_num >= length(SSjob.squads))
+		SSjob.active_squads = SSjob.squads
+		return TRUE
+	if(max_squad_num == 0)
+		return TRUE
+	var/list/preferred_squads = shuffle(SSjob.squads)
+	for(var/s in SSjob.squads)
+		preferred_squads[s] = 1
+	if(!length(preferred_squads))
+		to_chat(world, "<span class='boldnotice'>Error, no squads found.</span>")
+		return FALSE
+	for(var/i in GLOB.new_player_list)
+		var/mob/new_player/player = i
+		if(!player.ready || !player.client?.prefs?.preferred_squad)
+			continue
+		var/squad_choice = player.client.prefs.preferred_squad
+		if(squad_choice == "None")
+			continue
+		if(!preferred_squads[squad_choice])
+			stack_trace("[player.client] has in its prefs [squad_choice] for a squad. Not valid.")
+			continue
+		preferred_squads[squad_choice]++
+	sortTim(preferred_squads, cmp=/proc/cmp_numeric_dsc, associative = TRUE)
+
+	preferred_squads.len = max_squad_num
+	for(var/s in preferred_squads) //Back from weight to type.
+		preferred_squads[s] = SSjob.squads[s]
+	SSjob.active_squads = preferred_squads.Copy()
+
+	return TRUE
+
+/datum/game_mode/proc/initialize_scales()
```

Comment:
Since this looks to be completely an SSjob proc, we could just move it there instead, and ensure its called at `pre_start()`?

Reply:
The issue is that it is redefined depending on the game mode.

Reply:
Turned the relevant part into an SSjob proc.